images/benchmarks/syscallbench/syscallbench.c

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stdio.h>
+#include <sys/time.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <getopt.h>
+
+static int loops = 10000000;
+
+static void show_usage(const char *cmd)
+{
+	fprintf(stderr, "Usage: %s [options]\n"
+		"-l, --loops <num>\t\t Number of syscall loops, default 10000000\n", cmd);
+	exit(1);
+}
+
+int main(int argc, char *argv[])
+{
+	struct timeval start, stop, diff;
```

Comment:
I try to avoid using the container's timer, as gVisor's can distort the results. Instead, I'd start this with Spawn (analogous to `docker -d`)  the intial command being a `sleep` call, and then exec this binary. This will look like code in the fsbench library:
https://github.com/google/gvisor/blob/1e4ebb3ab826b6856b4b2360c4406f9726c7e44a/test/benchmarks/fs/fsbench/fsbench.go#L120

You should call spawn outside of the b.N loop.

Yes, you pay the cost of a `fork` here, but that overhead will diminish as b.N grows.

Reply:
ok, updated